### PR TITLE
Removed some dead code

### PR DESCRIPTION
**Pull request by BackEndTea**
This PR cleans up some dead code.

* MemoryUsageAware was moved to a package
* The exception wasn't used anymore
* Some unused variables

### CODE REVIEW

File: src/Logger/DebugFileLogger.php [hunk 1]
```diff
@@ -123,7 +123,7 @@ private function getResultsLine(
 
         $separateSections = false;
 
-        foreach ($executionResults as $index => $executionResult) {
+        foreach ($executionResults as $executionResult) {
```
Comment (theofidry) [inline src/Logger/DebugFileLogger.php:126]: may be an idea of a mutation?

**Comment by theofidry** [comment]
@maks-rafalko I think the failures are a bit weird, I cannot reproduce them locally with no cache enabled

File: src/Logger/DebugFileLogger.php (hunk 1 above)
Comment (maks-rafalko) [reply src/Logger/DebugFileLogger.php:126]: this mutation will in most cases be killed because of "undefined variable", no?

Comment (sanmai) [reply src/Logger/DebugFileLogger.php:126]: That, unless `$index` in unused.

Comment (theofidry) [reply src/Logger/DebugFileLogger.php:126]: Yeah would be nice to find a way to not blindly do it which would create too many errors. So a potential idea to explore but may not be doable

**Pull request merged**